src/test/java/uk/gov/beis/els/apiintegration/RateLimitingIntegrationTest.java

```diff
@@ -31,41 +38,101 @@
   @Autowired
   protected Bucket4JRequestService bucket4JRequestService;
 
-  @Before
-  public void setUp() throws Exception {
-    mockMvc = MockMvcBuilders
-        .webAppContextSetup(context)
-        .build();
+  @Test
+  public void withSingleIpAddress() throws Exception {
+    mockMvc.perform(post("/api/v1/domestic-ovens/gas-ovens/energy-label")
+            .contentType(MediaType.APPLICATION_JSON)
+            .content(getApiContent())
+            .accept(MediaType.APPLICATION_JSON)
+            .header("X-Forwarded-For", "1.1.1.1"))
+        .andExpect(status().isOk())
+        .andExpect(header().longValue("X-Rate-Limit-Remaining", 2));
   }
 
-  //TODO fix integration test: raised ELG-39 for this
   @Test
-  public void withSingleIpAddress() throws Exception {
+  public void withMultipleIpAddress() throws Exception {
+    mockMvc.perform(post("/api/v1/domestic-ovens/gas-ovens/energy-label")
+            .contentType(MediaType.APPLICATION_JSON)
+            .content(getApiContent())
+            .accept(MediaType.APPLICATION_JSON)
+            .header("X-Forwarded-For", "1.1.1.1, 2.2.2.2"))
+        .andExpect(status().isOk())
+        .andExpect(header().longValue("X-Rate-Limit-Remaining", 2));
```

Comment:
Think its worth testing that only the last IP counts towards the rate limit, e.g. just after this also call the below to ensure that even thought the first IP has changed the limit has still gone down
```
mockMvc.perform(post("/api/v1/domestic-ovens/gas-ovens/energy-label")
            .contentType(MediaType.APPLICATION_JSON)
            .content(getApiContent())
            .accept(MediaType.APPLICATION_JSON)
            .header("X-Forwarded-For", "3.3.3.3, 2.2.2.2"))
        .andExpect(status().isOk())
        .andExpect(header().longValue("X-Rate-Limit-Remaining", 1));
```